week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -60,13 +60,43 @@ var restaurantFinderApplication = {
     applicationVersion: "1.0",
     restaurants: restaurants,
     findAvailableRestaurants: function(numberOfPeople) {
+       
+        return restaurants.filter(getAvailableRestaurants).map(getName);
+
+        function getAvailableRestaurants(restaurant){
+            return (restaurant.totalSeats-restaurant.numberOfCustomers)>=numberOfPeople;
+        }
+
+        function getName(restaurant){
+            return restaurant.name;
+        }     
+     
+     
         // Complete here
     },
     findRestaurantServingDish: function(dishName) {
+
+        return restaurants.filter(getRestaurantDish).map(getName);
+
+        function getRestaurantDish(restaurant){
+            return restaurant.menu.includes(dishName);
+        }
+
+        function getName(restaurant) {
+            return restaurant.name;
+
         // Complete here
-    },
+    }
+},
     countNumberOfRestaurantsInArea: function(area) {
-        // Complete here
+       
+        var restaurantsInArea=restaurants.filter(getNumberOfRestaurant);
+
+        function getNumberOfRestaurant(restaurant){
+            return (restaurant.address.area===area);
+        }
```

Comment:
Again, this function doesn't return a number, so is badly named.  It looks like it returns true or false, and it takes a parameter to test.  So it's probably a predicate function.  How about `isInTargetArea`?  Then you'd say `restaurants.filter(isInTargetArea)`